backend/src/main/java/kernel360/techpick/feature/domain/pick/service/PickServiceImpl.java

```diff
@@ -1,100 +1,131 @@
 
 import org.apache.commons.lang3.ObjectUtils;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import kernel360.techpick.core.model.folder.Folder;
+import kernel360.techpick.core.model.folder.FolderType;
 import kernel360.techpick.core.model.pick.Pick;
+import kernel360.techpick.core.model.tag.Tag;
+import kernel360.techpick.feature.domain.folder.exception.ApiFolderException;
 import kernel360.techpick.feature.domain.pick.dto.PickCommand;
 import kernel360.techpick.feature.domain.pick.dto.PickMapper;
 import kernel360.techpick.feature.domain.pick.dto.PickResult;
+import kernel360.techpick.feature.domain.pick.exception.ApiPickException;
+import kernel360.techpick.feature.domain.tag.exception.ApiTagException;
 import kernel360.techpick.feature.infrastructure.folder.FolderAdaptor;
-import kernel360.techpick.feature.infrastructure.link.writer.LinkWriter;
 import kernel360.techpick.feature.infrastructure.pick.PickAdaptor;
-import kernel360.techpick.feature.infrastructure.user.reader.UserReader;
+import kernel360.techpick.feature.infrastructure.tag.TagAdaptor;
 import lombok.RequiredArgsConstructor;
 
 @Service
 @RequiredArgsConstructor
 public class PickServiceImpl implements PickService {
-	private final UserReader userReader; // TODO: change to UserAdaptor
-	private final LinkWriter linkWriter;
+
+	private final TagAdaptor tagAdaptor;
 	private final PickAdaptor pickAdaptor;
 	private final PickMapper pickMapper;
 	private final FolderAdaptor folderAdaptor;
 
 	@Override
 	@Transactional(readOnly = true)
 	public PickResult getPick(PickCommand.Read command) {
-		var user = userReader.readUser(command.userId());
-		var pick = pickAdaptor.getPick(user, command.pickId());
-		return pickMapper.toReadResult(pick);
+		checkPickUserAuthorization(command.userId(), command.pickId());
+		var pick = pickAdaptor.getPick(command.pickId());
+		return pickMapper.toPickResult(pick);
+	}
+
+	@Override
+	@Transactional(readOnly = true)
+	public PickResult getPickUrl(Long userId, String url) {
+		Pick pick = pickAdaptor.getPickUrl(userId, url);
+		if (ObjectUtils.notEqual(userId, pick.getUser().getId())) {
+			throw ApiPickException.PICK_UNAUTHORIZED_ACCESS();
+		}
+		return pickMapper.toPickResult(pick);
 	}
 
 	@Override
 	@Transactional
 	public PickResult saveNewPick(PickCommand.Create command) {
-		var user = userReader.readUser(command.userId());
-		var folder = folderAdaptor.getFolder(command.parentFolderId());
-		var link = linkWriter.writeLink(command.linkInfo());
-		var pick = pickAdaptor.savePick(pickMapper.toEntity(command, user, folder, link));
-		return pickMapper.toCreateResult(pick);
+		checkFolderUserAuthorization(command.userId(), command.parentFolderId());
+		var pick = pickAdaptor.savePick(command);
+		pick.getParentFolder().getChildPickOrderList().add(pick.getId());
+
+		List<Long> tagOrderList = pick.getTagOrderList();
+		List<Tag> tagList = tagAdaptor.getTagList(tagOrderList);
```

Comment:
p3: savePickTag는 DB관계 테이블에 의한 함수라 어뎁터 내부로 빼는게 어떨지.

```suggestion
    checkFolderUserAuthorization(command.userId(), command.parentFolderId());
    checkTagListUserAuthorization(command.userId(), command.tagIdList())
    var pick = pickAdator.savePick(command); // savePick에서 아예 pickTag 관계도 함께 설정
    return pickMapper.toPickResult(pick);
```